packages/admin/cms-admin/src/dam/FileForm/ReplaceFileButton.tsx

```diff
@@ -0,0 +1,98 @@
+import { useApolloClient } from "@apollo/client";
+import { useErrorDialog } from "@comet/admin";
+import { ThreeDotSaving, Upload } from "@comet/admin-icons";
+import { Button } from "@mui/material";
+import axios, { CancelTokenSource } from "axios";
+import { useRef, useState } from "react";
+import { FileRejection, useDropzone } from "react-dropzone";
+import { FormattedMessage } from "react-intl";
+
+import { useCmsBlockContext } from "../../blocks/useCmsBlockContext";
+import { replaceById } from "../../form/file/upload";
+import { convertMimetypesToDropzoneAccept } from "../DataGrid/fileUpload/fileUpload.utils";
+import { DamFileDetails } from "./EditFile";
+
+interface ReplaceFileButtonProps {
+    file: DamFileDetails;
```

Comment:
Just noticed: It isn't really a good idea that all components surrounding EditFile use the same GraphQL fragment. We should move the fragments to the components in a follow-up PR to follow our recommended colocating fragments strategy.

Reply:
https://vivid-planet.atlassian.net/browse/COM-1322